src/core/game.js

```diff
@@ -0,0 +1,184 @@
+const SpaceShip = require("../core/spaceShip");
+const Point = require("../utils/point");
+const mathSupport = require("../utils/mathSupport");
+const Galaxy = require("./galaxy");
+
+class Game {
+    constructor(config) {
+        this.gameConfig = config;
+        this.player = new SpaceShip(new Point(3, 3), new Point(3, 3), config.MAX_POWER);
+        this.galaxy = new Galaxy(config.GALAXY);
+        this.isOver = false;
+        this.starDates = config.INITIAL_STARDATES;
+        this.knownGalaxy = [];
+        for (let i = 0; i < Math.sqrt(config.GALAXY.QUADRANTS); i++) {
+            this.knownGalaxy.push([]);
+            for (let j = 0; j < Math.sqrt(config.GALAXY.QUADRANTS); j++) {
+                this.knownGalaxy[i].push(["?", "?", "?"]);
+            }
+        }
+    }
+
+    makeKlingonsTurn() {
+        this.galaxy.removeDestroyed();
+        this.galaxy.getQuadrantObjects(this.player.quadrant).forEach((so, i) => {
+            if (so.type == "ship" ) {
+                let distanceToPlayer = mathSupport.cityBlockDistance(
+                    this.player.sector.x,
+                    this.player.sector.y,
+                    so.sector.x,
+                    so.sector.y
+                );
+                let dmg = 30;
+                so.reducePower(dmg);
+                let hasHit = Math.random() < (5 / (distanceToPlayer + 4));
+                if(hasHit){
+                    console.log(`Klingons has attacked and hit you! You lose ${dmg} power.`);
+                    this.player.reducePower(dmg);
+                } else {
+                    console.log('Klingons has attacked, but missed.');
+                }
+            }
+        });
+    }
+
+    playerShoot(power){
+        if(power >= this.player.power){
+            console.log('You do not have enough power!');
+            return;
+        }
+        let closestKlingons = this.galaxy.getQuadrantObjects(this.player.quadrant)
+            .filter(so => so.type == 'ship')
+            .sort((prev, cur) => {
+                let prevDist = mathSupport.cityBlockDistance(this.player.sector.x, this.player.sector.y, prev.sector.x, prev.sector.y);
+                let curDist = mathSupport.cityBlockDistance(this.player.sector.x, this.player.sector.y, cur.sector.x, cur.sector.y);
+                return prevDist > curDist;
+            });
+        if(closestKlingons.length <= 0){
+            console.log('There is no remaining klingons in your current quadrant');
```

Comment:
I would add `return` statement here and get rid of `else` statement below. I like the approach with sticking to the rule: quick ASAP. Basically return from the function as fast as you can. If you can.